src/topic_based_system.cpp

```diff
@@ -163,6 +163,17 @@ CallbackReturn TopicBasedSystem::on_init(const hardware_interface::HardwareInfo&
     ready_to_send_cmds_ = false;
   }
 
+  // Check if return of get_hardware_parameter is double type. If not, turn it from string to double type
+  try
+  {
+    block_joint_command_threshold_ = std::stod(get_hardware_parameter("block_joint_command_threshold", "10.0"));
```

Comment:
Should we use 5 as default value? Following the original code? Or U have some other thoughts?